neil

```diff
@@ -411,6 +411,57 @@ options can be used to control the add-deps behavior:
               (println "[neil] First execute: neil add test")))
           (println "[neil] Not inside a deps.edn project."))))))
 
+(ns babashka.neil.version
+  (:require [babashka.fs :as fs]
+            [babashka.neil.meta :as meta]
+            [babashka.neil.project :as proj]
+            [clojure.edn :as edn]
+            [clojure.spec.alpha :as s]
```

Comment:
I prefer to leave clojure.spec.alpha out of the equation here for now, since this incurs some startup time to load. The s/def below seems pretty trivial to do without spec.

Reply:
Btw, this can also done with bb.cli `:validate`, let's put it there?

Reply:
@borkdude:
> I prefer to leave clojure.spec.alpha out of the equation here for now, since this incurs some startup time to load. The s/def below seems pretty trivial to do without spec.

Yeah, I was on the fence about this since it's an external dep. We can replace the spec with a regular predicate function.

> Btw, this can also done with bb.cli `:validate`, let's put it there?

In this case we aren't validating CLI options, we're validating the data inside the `deps.edn` file.

Reply:
I'm moving the logic to a `project-version?` predicate so we can use it in `:validate` when we add the other subcommands.

Reply:
Actually this particular predicate might not make sense there, but I'll keep `:validate` in mind for the future.